### ISSUE TITLE
New Feature for packaging configuration

### ISSUE THREAD

**Issue by dsugden** [sbt/sbt-bundle#42]
With Regards to the possibility of:

having a convention of a folder structure:
 
    src/main/configuration/<name>

for some configuration containing some files, and then introducing the Task:

    config:dist

to package this.

I have a question. I've tried to:

    NativePackagerKeys.dist in Config :=  Def.taskDyn {
         createConfig.evaluated
    }.value,

where createConfig is a Def.inputTask

I get a 

    `value` can only be called on an input task within a task definition macro, such as := or Def.inputTask

error on compilation. It would seem this isn't possible  (InputTask[T]  => TaskKey[T])

http://stackoverflow.com/questions/28590296/using-input-args-inside-a-taskkey

My Question is:

is it reasonable to introduce a Command:    >  config <foldername>

which would set State so config:dist would work?

so the user would:

>sbt
>config frontend
>config:dist

Thoughts?

**Comment by huntc** [comment]
You don't want a command. You'll need to define both `stage` and `dist` e.g.:

```scala
    NativePackagerKeys.dist in Configuration := Def.taskDyn {
      Def.task {
        createConfiguration()
      }.value
    }.value,
    NativePackagerKeys.stage in Configuration := Def.taskDyn {
      Def.task {
        stageConfiguraiton()
      }.value
    }.value,
```

`Configuration` is defined in a similar way to `Bundle`:

```
  val Configuration = config("configuration") extend Universal
```

Your functions are then like this:

```scala
  private def createConfiguration(): Def.Initialize[Task[File]] = Def.task {
    val configurationTarget = (target in Configuration).value
```

etc. HTH.

**Comment by dsugden** [comment]
@huntc  

I have something working https://github.com/dsugden/sbt-bundle/blob/configDist/src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala



This (prototype) works for a configuration called "frontend"  located at src/main/configuration/frontend

It produces:

    > config:dist
    [info] Configuration Bundle has been created: /Users/dave/projects/akka/sbt-bundle/sbt-bundle-          tester/target/configuration/frontend-db887283d431661593cf4d48275d7628a7a74cb30da4e4dd62362a5a189a05c9.zip


I have some questions before I continue :

I'm using the stage task inside the createConfiguration() :

    val configurationTarget = (NativePackagerKeys.stage in Configuration).value

to use as the source for bundleMappings... ok? Should this be cleaned up (deleted)?

Is the expectation that this task:
 
    config:dist

will create zips for ALL the folder contained in src/main/configuration ? Or that the user will somehow specify which configuration to target. What would that mechanism look like? You said no command, so maybe a settings key?

Creating zips for each subfolder of configuration is fine as well... the  tab completion / parsing in sbt conductr could auto-fill or present a list?

?

thanks

**Comment by dsugden** [comment]
bump ^  :)

**Comment by huntc** [comment]
Ah, sorry. I'll take a look first thing.

**Comment by huntc** [comment]
This may be easier for me to feedback on if you create a PR now, but with the title prefixed "WIP - DO NOT MERGE - ".

**Comment by huntc** [comment]
Fixed by #47
